packages/elements/src/react/sign-in/action/action.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+
+import type { FormSubmitProps } from '~/react/common';
+import { Submit } from '~/react/common';
+
+import { SignInNavigate, type SignInNavigateElementKey, type SignInNavigateProps } from './navigate';
+import { SignInResend, type SignInResendProps } from './resend';
+
+export type SignInActionProps = { asChild?: boolean } & FormSubmitProps &
+  (
+    | ({
+        navigate: SignInNavigateProps['to'];
+        resend?: never;
+        submit?: never;
+      } & Omit<SignInNavigateProps, 'to'>)
+    | { navigate?: never; resend?: never; submit: true }
+    | ({ navigate?: never; resend: true; submit?: never } & SignInResendProps)
+  );
+
+export type SignInActionCompProps = React.ForwardRefExoticComponent<
+  Exclude<SignInActionProps, 'navigate'> & {
+    to: SignInNavigateElementKey;
+  } & React.RefAttributes<HTMLButtonElement>
+>;
+
+const SIGN_IN_ACTION_NAME = 'SignInAction';
+
+/**
+ * Perform various actions during the sign-in process. This component is used to navigate between steps, submit the form, or resend a verification codes.
+ *
+ * @example
+ * import { Action } from '@clerk/elements/sign-in';
+ * <Action navigate="start">Go Back</Action>
+ *
+ * @example
+ * import { Action } from '@clerk/elements/sign-in';
+ * <Action submit>Sign In</Action>
+ *
+ * @example
+ * import { Action } from '@clerk/elements/sign-in';
+ * <Action resend>Resend</Action>
+ */
+export const SignInAction = React.forwardRef<React.ElementRef<'button'>, SignInActionProps>((props, forwardedRef) => {
```

Comment:
Are these changes related to the reset password flow? I think not, just clarifying that it's a refactor 👀

Reply:
It is a part of a refactor. Just moving the files into a more logical file structure. Largely just grouping Steps and Actions.